src/xpk/commands/cluster.py

```diff
@@ -696,6 +697,146 @@ def cluster_create_ray_cluster(args) -> None:
+  3. Retrieves Google Kubernetes Engine (GKE) cluster credentials.
+  4. Deploys CoreDNS to the cluster.
+  5. Scales down the 'kube-dns-autoscaler' and 'kube-dns' deployments.
+  6. Scales up the 'coredns' deployment to 15 replicas.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  home_dir = os.path.expanduser("~")
+  coredns_repo_dir_name = "deployment" 
+  coredns_repo_full_path = os.path.join(home_dir, coredns_repo_dir_name)
+  coredns_k8s_path = os.path.join(coredns_repo_full_path, "kubernetes")
+
+  command_jq_install = (
+    f'sudo apt install jq -y'
+  )
+  return_code = run_command_with_updates(
+        command_jq_install, 'Install jq', args
+    )
+  if return_code != 0:
+        xpk_print(f'[XPK] Install jq error {return_code}')
+        xpk_exit(return_code)
+
+  # Check if the target directory already exists to avoid errors caused by duplicate cloning
+  if os.path.exists(coredns_repo_full_path):
+    xpk_print(f"[XPK] Directory '{coredns_repo_full_path}' already exists, skip git clone.")
+  else:
+    command_git_clone = (
+      f'git clone https://github.com/coredns/deployment.git {coredns_repo_full_path}'
+    )
+    xpk_print(f"[XPK] Task: 'Clone deployment ' in progress, Target directory:{coredns_repo_full_path}.")
+    return_code = run_command_with_updates(
+      command_git_clone, 'Clone deployment', args
+    )
+    if return_code != 0:
+      xpk_print(f'[XPK] Clone deployment error {return_code}')
+      xpk_exit(return_code)
+
+
+  cluster_name = os.environ.get('CLUSTER_NAME')
+  region = os.environ.get('REGION')
+  project = os.environ.get('PROJECT')          
+
+  if not all([cluster_name, region, project]):
+    missing_vars = []
+    if not cluster_name: missing_vars.append('CLUSTER_NAME')
+    if not region: missing_vars.append('REGION')
+    if not project: missing_vars.append('PROJECT')
+    xpk_print(f"[XPK] Error: Missing required environment variable:{', '.join(missing_vars)}. Please ensure these variables are set.")
+    xpk_exit(1)
+
+  command_get_credentials = (
+    f'gcloud container clusters get-credentials {cluster_name} --dns-endpoint '
+    f'--region={region} --project={project} && kubectl config view '
+    f'&& kubectl config set-context --current --namespace=default'
+  )
+  xpk_print(f"[XPK] Task: 'Get cluster credentials' in progress.")
+  return_code = run_command_with_updates(
+    command_get_credentials, 'Get cluster credentials', args
+  )
+  if return_code != 0:
+    xpk_print(f'[XPK] Failed to get cluster credentials {return_code}')
+    xpk_exit(return_code)
+
+  if not os.path.isdir(coredns_k8s_path):
+    xpk_print(f"[XPK] Error：CoreDNS Kubernetes path '{coredns_k8s_path}' does not exist. Has git clone been successful?")
+    xpk_exit(1)
+
+  # Remember the current directory so that you can restore it later
+  original_cwd = os.getcwd()
```

Comment:
Curious, can we execute the script from the current directory and just point to it rather than cd-ing to it? E.g.`./{path_to_core_dns}/deploy.sh | kubectl apply -f -`. If not, then this may be ok since the latter `kubectl apply -f -` may not be possible?